be/src/exec/jni_scanner.cpp

```diff
@@ -58,30 +58,38 @@ Status JniScanner::do_open(RuntimeState* state) {
 void JniScanner::do_update_counter(HdfsScanProfile* profile) {}
 
 void JniScanner::do_close(RuntimeState* runtime_state) noexcept {
-    if (_jni_scanner_obj == nullptr) return;
     JNIEnv* _jni_env = JVMFunctionHelper::getInstance().getEnv();
-    _jni_env->CallVoidMethod(_jni_scanner_obj, _jni_scanner_close);
-    _check_jni_exception(_jni_env, "Failed to close the off-heap table scanner.");
-    _jni_env->DeleteLocalRef(_jni_scanner_obj);
-    _jni_env->DeleteLocalRef(_jni_scanner_cls);
+    if (_jni_scanner_obj != nullptr) {
+        if (_jni_scanner_close != nullptr) {
+            _jni_env->CallVoidMethod(_jni_scanner_obj, _jni_scanner_close);
```

Comment:
Set close to nullptr?

Reply:
well, No need to. 

set _jni_scanner_obj/_jni_scanner_cls to nullptr is also kind of unecessary either.